src/main/java/duke/tasks/TaskList.java

```diff
@@ -0,0 +1,306 @@
+package duke.tasks;
+
+import java.util.ArrayList;
+
+import duke.exceptions.DukeIOException;
+import duke.exceptions.DukeIllegalArgumentException;
+import duke.storage.Storage;
+
+/**
+ * A list of tasks.
+ */
+public class TaskList {
+    // Error messages
+    private static final String ERROR_MESSAGE_INDEX_OUT_OF_BOUNDS =
+            "The task number is out of range. Use \"list\" to see your tasks.";
+
+    // Message Templates
+    private static final String ADD_TASK_TEMPLATE =
+            "Got it. I've added this task:%n%s%nNow you have %d %s in the list.";
+    private static final String DELETE_TASK_TEMPLATE =
+            "Noted. I've removed this task:%n%s%nNow you have %d %s in the list.";
+
+    // An ArrayList that stores the list of tasks.
+    protected final ArrayList<Task> list;
```

Comment:
This wasn't the most intuitive name to me... 
Perhaps a more descriptive name than 'list' could improve readability?

Reply:
Got it, thank you for pointing it out!